installers/agent/release/agent.sh

```diff
@@ -105,21 +143,18 @@ export AGENT_STARTUP_ARGS
 export LOG_DIR
 export LOG_FILE
 
-CMD="$JAVA_HOME/bin/java -jar \"$AGENT_DIR/agent-bootstrapper.jar\" $GO_SERVER $GO_SERVER_PORT"
+RUN_CMD=("$(autoDetectJavaExecutable)" "-jar" "$AGENT_DIR/agent-bootstrapper.jar" "$GO_SERVER" "$GO_SERVER_PORT")
 
-echo "[`date`] Starting Go Agent Bootstrapper with command: $CMD" >>"$STDOUT_LOG_FILE"
-echo "[`date`] Starting Go Agent Bootstrapper in directory: $AGENT_WORK_DIR" >>"$STDOUT_LOG_FILE"
-echo "[`date`] AGENT_STARTUP_ARGS=$AGENT_STARTUP_ARGS" >>"$STDOUT_LOG_FILE"
+echo "[$(date)] Starting Go Agent Bootstrapper with command: ${RUN_CMD[@]}" >>"$STDOUT_LOG_FILE"
+echo "[$(date)] Starting Go Agent Bootstrapper in directory: $AGENT_WORK_DIR" >>"$STDOUT_LOG_FILE"
+echo "[$(date)] AGENT_STARTUP_ARGS=$AGENT_STARTUP_ARGS" >>"$STDOUT_LOG_FILE"
 cd "$AGENT_WORK_DIR"
 
-if [ "$JAVA_HOME" == "" ]; then
-    echo "Please set JAVA_HOME to proceed."
-    exit 1
-fi
 
 if [ "$DAEMON" == "Y" ]; then
-    eval exec nohup "$CMD" >>"$STDOUT_LOG_FILE" 2>&1 &
+    "${RUN_CMD[*]}"  >> "$STDOUT_LOG_FILE" 2>&1 &
```

Comment:
Shouldn't this be `${RUN_CMD[@]}`? Here's what happens on my shell currently:

```
$ ABC=("ls" "-l")
$ "${ABC[*]}"
-bash: ls -l: command not found
$ "${ABC[@]}"
total 44
-rw-r--r--  1 aravindsv staff  1196 Apr 20 23:11 CODE_OF_CONDUCT.md
-rw-r--r--  1 aravindsv staff   538 Apr 21 08:52 CONTRIBUTING.md
-rw-r--r--  1 aravindsv staff   623 Apr 21 08:52 Gemfile
-rw-r--r--  1 aravindsv staff  4161 Apr 21 08:52 Gemfile.lock
-rw-r--r--  1 aravindsv staff 11358 Apr 21 08:52 LICENSE
-rw-r--r--  1 aravindsv staff  1509 Apr 21 08:52 README.md
-rw-r--r--  1 aravindsv staff   183 Apr 20 23:11 Rakefile
drwxr-xr-x 52 aravindsv staff  1768 Apr 23 05:22 build
-rw-r--r--  1 aravindsv staff  3322 May  3 07:13 config.rb
drwxr-xr-x  9 aravindsv staff   306 Apr 20 23:11 data
drwxr-xr-x  4 aravindsv staff   136 Apr 20 23:11 lib
drwxr-xr-x 31 aravindsv staff  1054 May  3 07:13 source
$ echo $IFS | od -xc
0000000    000a
         \n
0000001
```

`${RUN_CMD[*]}` will make the whole thing one param. You'll need to use eval if you really want to use `*`.